prov/gni/src/gnix_init.c

```diff
@@ -150,6 +151,15 @@ void _gnix_init(void)
 	static int called=0;
 
 	if (called==0) {
+		if (sizeof(struct gnix_mr_key) != 8) {
```

Comment:
Runtime check to ensure the memory registration key is the size we expect it to be. It should be 64 bits, or the size of a uint64_t. 

Sloppiness on my part, but it needs be the same size as the returned value from fi_mr_key